lib/pdf_attachment_reporter.rb

```diff
@@ -0,0 +1,177 @@
+#  Notes/limitations of this report
+#
+#  - We do not take into account changes in ownership of a document with
+#  a PDF attachment over time. Instead we take the most recently-published
+#  version of the document as the correct source of ownership.
+#
+#  - A document that has had a PDF published in the past under one name that
+#  is subsequently replaced by a PDF with another name is counted in the
+#  report twice. Even though the site shows only one PDF.
+#
+#  - All policy group attachments are lumped into one category
+#
+#  - Finding the organisation which owns a PDF can be complex. In
+#  cases where there are multiple owners, we first try to find a parent
+#  organisation, if we know that all associated organisations are related.
+#  Otherwise we assume that the first organisation in the list is a best
+#  guess.
+
+require 'csv'
+
+class PDFAttachmentReporter
+  POLICY_GROUPS = 'Policy Groups'.freeze
+
+  def initialize(opts = {})
+    @data_path = opts.fetch(:data_path, ENV['HOME'])
+    @first_period_start_date = opts.fetch(:first_period_start_date, Date.parse('2016-01-01'))
+    @last_time_period_days = opts.fetch(:last_time_period_days, 30)
+  end
+
+
+  def pdfs_by_organisation
+    second_time_period_date = @last_time_period_days.days.ago.to_date
+
+    live_organisation_names = Organisation.where(govuk_status: 'live').map(&:name) << POLICY_GROUPS
+
+    live_organisation_published_pdfs_total_counts_hash = Hash[live_organisation_names.map { |o| [o, 0] }]
+    live_organisation_published_pdfs_since_first_period_counts_hash = Hash[live_organisation_names.map { |o| [o, 0] }]
+    live_organisation_published_pdfs_since_second_period_counts_hash = Hash[live_organisation_names.map { |o| [o, 0] }]
+
+    unique_published_pdf_attachments.each do |attachment|
```

Comment:
Because of the `group` here, in order to get an overall count for the total for a progress bar, you'd need something like `scope.count.keys.count`.